### ISSUE TITLE
Flaky E2E: Select a free .wordpress.com domain

### ISSUE THREAD

**Issue by worldomonation** [Automattic/wp-calypso#73449]
### Spec file

specs/plans/plans__signup-free.ts

### TeamCity ID

9501190

### Logs

```
page.waitForSelector: Timeout 10000ms exceeded.
=========================== logs ===========================
waiting for locator('.domain-suggestion__content:has-text(".wordpress.com")') to be visible
============================================================
at DomainSearchComponent.selectDomain (/home/teamcity-2/buildAgent/work/c4a9d5b38c1dacde/packages/calypso-e2e/src/lib/components/domain-search-component.ts:65:38)
    at Object.<anonymous> (/home/teamcity-2/buildAgent/work/c4a9d5b38c1dacde/test/e2e/specs/plans/plans__signup-free.ts:60:54)
```

**Comment by worldomonation** [comment]
![image](https://user-images.githubusercontent.com/6549265/219462437-1ee86e4b-1783-42f0-8364-e0166fdd2a3c.png)

**Comment by worldomonation** [comment]
[plans__signup-free__select-a-free--wordpress-com-domain__1-1.webm](https://user-images.githubusercontent.com/6549265/219462499-57db8c38-0be1-428e-ac73-3fa57c07f466.webm)

**Comment by worldomonation** [comment]
If we can deal with the external domain lookup issue at the domain selection stage (whether during signup or launch) we can significantly cut down instances of flaky failures.

**Comment by worldomonation** [comment]
Should be fixed by #73454.
